test/ResourceGenerator/DoctrinePaginatorTest.php

```diff
@@ -48,11 +48,11 @@ public function setUp(): void
     }
 
     /**
-     * @psalm-return AbstractQuery&MockObject
+     * @psalm-return Query&MockObject
      */
-    public function mockQuery(): AbstractQuery
+    public function mockQuery(): Query
```

Comment:
```suggestion
    public function mockQuery(): Query&MockObject
```

May as well make this an intersection whilst we're here.

Reply:
Acutally, can't this entire method be replaced with `$this->createMock(Query::class)`